site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -230,14 +229,23 @@ class SitePermissionsDialogActivityLauncher @Inject constructor(
             layout.setPadding(0, 0, 0, 40.toPx())
         }
         snackbar.apply {
-            setAction(R.string.sitePermissionsDeniedSnackBarAction) {
-                onPermissionAllowed()
-            }
+            setAction(R.string.sitePermissionsDeniedSnackBarAction) { onPermissionAllowed() }
+            addCallback(
+                object : Snackbar.Callback() {
+                    override fun onDismissed(
+                        transientBottomBar: Snackbar?,
+                        event: Int,
+                    ) {
+                        if (event == BaseCallback.DISMISS_EVENT_TIMEOUT) { sitePermissionRequest.deny() }
```

Comment:
This is working as expected. 

Minor nit: do we need to specifically check for a timeout event, or could we just call `sitePermissionRequest.deny()` when `onDismissed()` is called? (thinking that while right now those snackbars are not user-dismissable, if we changed that elsewhere in the code we'd suddenly miss that use case and wouldn't call deny if the user swiped the snackbar away)

Reply:
@CDRussell this is on purpose because the user can select the onAction button in the snackbar which will launch the system permissions dialogs again. Therefore, if I `deny()` the permission in `onDismissed()` I won't be able to grant it after user selects allowing system permission in this second attempt.

Reply:
Makes sense!